api/tools/src/main/java/org/pmiops/workbench/tools/ImpersonatedServiceAccountApiClientFactory.java

```diff
@@ -0,0 +1,43 @@
+package org.pmiops.workbench.tools;
+
+import com.google.cloud.iam.credentials.v1.IamCredentialsClient;
+import com.google.protobuf.Duration;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.firecloud.ApiClient;
+
+public class ImpersonatedServiceAccountApiClientFactory extends ApiClientFactory {
+
+  public ImpersonatedServiceAccountApiClientFactory(
+      String targetServiceAccount, WorkbenchConfig config) {
+    try {
+      this.apiClient = newApiClient(targetServiceAccount, config);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public ApiClient newApiClient(String targetServiceAccount, WorkbenchConfig config)
+      throws IOException {
+    IamCredentialsClient iamCredentialsClient = IamCredentialsClient.create();
+    List<String> delegates =
+        Arrays.asList("projects/-/serviceAccounts/" + config.auth.serviceAccountApiUsers.get(0));
```

Comment:
This config value is effectively an allowlist of service accounts that are allowed to call the RW API. There is nothing special about the first element of the array, so this is not a reliable way of determining the appspot account email. Why are the delegates even needed here?

Reply:
I thought I needed it from what the API documentation says but you're right. It works even when I leave it out.